src/app/COMPONENTS/task/task.component.ts

```diff
@@ -27,25 +26,17 @@ import { BehaviorSubject, Observable } from 'rxjs';
 
 
 export class TaskComponent {
-  availableTasks$ = new BehaviorSubject<Task[]>([])
 
-  constructor(private taskService: TasksService) {
-  }
-  ngOnInit() {
-    this.taskService.getAllTasks().subscribe(data => {
-      this.availableTasks$.next(data)
+  availableTasks$ = this.taskService.getAllTasks()
 
-    })
+  constructor(private taskService: TasksService) {
   }
 
   deleteTask(taskId: string) {
-    this.taskService.deleteTask(taskId)
-    const updatedTasksArray = this.availableTasks$.getValue().filter((task) => task.id !== taskId);
-    this.availableTasks$.next(updatedTasksArray)
-
+    this.availableTasks$ = this.taskService.deleteTask(taskId)
   }
   taskDetails(taskId: string) {
-    this.taskService.taskDetails(taskId)
+    return this.taskService.taskDetails(taskId)
```

Comment:
taskService.taskDetails doesn't return anything, so we don't need a return statement here